packages/client/hooks/useTrebuchetEvents.ts

```diff
@@ -55,16 +55,34 @@ const useTrebuchetEvents = () => {
 
     const onReconnected = () => {
       setConnectedStatus(true)
-      atmosphere.retries.forEach((retry) => retry())
-      atmosphere.eventEmitter.emit('removeSnackbar', ({key}) => key === 'offline')
     }
 
     const onData = async (payload: string | object) => {
       // hacky but that way we don't have to double parse huge json payloads. SSE graphql payloads are pre-parsed
       const obj = typeof payload === 'string' ? JSON.parse(payload) : payload
-      if (obj.version && obj.version !== __APP_VERSION__ && 'serviceWorker' in navigator) {
-        const registration = await navigator.serviceWorker.getRegistration()
-        registration?.update().catch()
+
+      if (obj.version && obj.version !== __APP_VERSION__) {
+        atmosphere.retries.clear()
+        atmosphere.eventEmitter.emit('addSnackbar', {
+          autoDismiss: 0,
+          message: 'You’re offline, Please refresh to continue',
```

Comment:
We may not want to show this snackbar if:
- We are sure that redeploy would always cause an extra small amount of downtime
- We are sure (or ignore) that in this extra small amount of downtime no data would become stale

Reply:
-1 I'm not sure if I get it correctly. When we're updating, after we reconnected to the server, we're now showing the you're offline message? What's the reasoning behind it?

Reply:
@Dschoordsch As far as I understand, we want to not refetch the queries, if reconnect occurred because of a version bump. 
As data might be outdated in this case, my proposal is to show the refresh button.
This will be shown only on socket reconnect and in case of version change. So the frontend version would be different from the backend. 
I'm actually not sure if It is really needed, we can remove this snackbar.

Reply:
What usually happens on update is that the frontend is reloaded and then it tries to connect to the backend, which is unavailable for some time. So I'm not even sure if that's the right condition, as in that case once both reconnect, the version should be identical, shouldn't it?

Reply:
@Dschoordsch Updated test cases to make it cleaner.
Removed the snackbar as discussed here: https://github.com/ParabolInc/parabol/issues/5548#issuecomment-991183359

How it currently works in master:

- When socket reconnects, frontend is not reloaded, we only refetch GraphQL queries
- When the backend receives a connection it sends an app version to the frontend.
- If this version is different we update service workers.

In my PR, If reconnect occurred, I refetch GraphQL queries only if the version didn't change.